### PR TITLE
Implement operator.contains for empty Tuples

### PR DESCRIPTION
**Pull request by brandonwillard**
This PR implements `x in ()`.

### CODE REVIEW

**Comment by esc** [comment]
@brandonwillard thank you for submitting this, I have added it to the queue for review. It seems like there are some public CI failures that you could perhaps look into in the meantime.

File: numba/cpython/tupleobj.py [hunk 1]
```diff
@@ -406,3 +406,8 @@ def tuple_index_impl(tup, value):
         raise ValueError("tuple.index(x): x not in tuple")
 
     return tuple_index_impl
+
+
+@overload(operator.contains, types.Tuple([]), types.Any)
```
Comment (stuartarchibald) [inline numba/cpython/tupleobj.py:411]: Providing a signature isn't yet implemented for `@overload`, you'll need to put the type checks into the function body. https://numba.readthedocs.io/en/stable/extending/overloading-guide.html is a good place to start learning about `@overload` and has examples patterns to follow. Due to the lack of guards on the types, this function is accepting any types and is most likely overriding other functions that do similar, as a result some test fail. Good news is that the fix should be quick!

Comment (brandonwillard) [reply numba/cpython/tupleobj.py]: Ha, yeah, I was thinking of a different decorator when I whipped that together.  I've updated it now.

**Comment by brandonwillard** [comment]
All right, after that change, I'm not sure what those errors are about.  There seems to be [some sort of timeout](https://dev.azure.com/numba/numba/_build/results?buildId=8615&view=logs&j=a506c6d0-8428-5bb5-7d36-c242a5467a36&t=3b81a997-a0fe-5621-27e6-18b6744841b9&l=9741) while attempting to obtain a compiler lock.

**Review by stuartarchibald — APPROVED** [review]
Thanks for the patch!

**Comment by stuartarchibald** [comment]
@brandonwillard Congratulations on your first contribution to Numba!

**Comment by brandonwillard** [comment]
My pleasure, and thanks for taking the time to look over this&mdash;especially given that wildly mistaken `overload`. 

Final thought/follow-up: would it actually be better to have an empty tuple inherit from `Sequence` instead?

**Comment by sklam** [comment]
@brandonwillard, the `Sequence` type implies iterable but due to the typing limitation we can't make mix-typed tuple to be iterable; e.g.:

```python
for val in (1, 2.3, "str"):   # type inference can't determine type of `val`
    ...
```

**Pull request merged**